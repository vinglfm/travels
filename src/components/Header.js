import React, {Component} from 'react';
import AppBar from '@material-ui/core/AppBar';
import Button from '@material-ui/core/Button';
import Toolbar from '@material-ui/core/Toolbar';
import Typography from '@material-ui/core/Typography';
import AuthModal from './Auth/AuthModal';
import actions from '../_actions';
import {connect} from 'react-redux';
import styles from './Header.css';

export class Header extends Component {
    constructor(props) {
        super(props);
        this.state = {
            openAuthModal: false
        };
        this.openAuthModal = this.openAuthModal.bind(this);
        this.closeAuthModal = this.closeAuthModal.bind(this);
    }

    openAuthModal() {
        this.setState({
            openAuthModal: true
        });
    }

    closeAuthModal() {
        this.setState({
            openAuthModal: false
        });
    }

    render() {
        return (
            <div className={styles.header__root}>
                <AppBar title='Travels'>
                    <Toolbar>
                        <Typography variant='title' color='inherit' className={styles.header__title}>
                            Travels
                        </Typography>
                        {this.props.requireLogIn ? (
                            <Button color='inherit' onClick={this.openAuthModal}>Sign In</Button>
                        ) : (
                            <Button color='inherit' onClick={this.props.logOut}>Log out</Button>
                        )}
                    </Toolbar>
                </AppBar>
                <AuthModal open={this.state.openAuthModal} handleClose={this.closeAuthModal}/>
            </div>
        );
    }
}

const mapStateToProps = (state, ownProps) => ({
    requireLogIn: Object.keys(state.user).length === 0 
  });

const mapDispatchToProps = {
    logOut: actions.logOut
};

export default connect(mapStateToProps, mapDispatchToProps)(Header);