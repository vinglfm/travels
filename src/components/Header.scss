:local .header {

    &__root {
        flex-grow: 1;
    }

    &__title {
        flex: 1;    
    }
}

:local .fb__button {
    font-family: Helvetica, sans-serif;
    font-weight: 500;
    font-size: 1.2rem;
    background-color: #4c69ba;
    color: white;
    border: 1px solid rgba(0, 0, 0, 0.1);
    cursor: pointer;
    display: inline-block;
    text-decoration: none;
    text-transform: uppercase;
    background-color: #4c69ba;
    padding: .6rem;
}

:local .lines {
    margin: 1rem;
    line-height: 0.5;
    text-align: center;
    font-family: Helvetica, sans-serif;
    font-weight: 500;
    font-size: 1.2rem;

    span {
        display: inline-block;
        position: relative;  

        &:before,
        &:after {
            content: "";
            position: absolute;
            height: 0.1rem;
            border-bottom: 1px solid black;
            border-top: 1px solid black;
            top: 50%;
            width: 3rem;
        }

        &:before {
            right: 100%;
            margin-right: 2px;
        }

        &:after {
            left: 100%;
            margin-left: 2px;
        }
    }
  }

:local .log_in {
    font-style: italic;
    text-align: center;
    a {
        font-weight: bold;
        font-size: 1.2rem;
        cursor: pointer;
    }
}

:local .sign__up {
  font-style: italic;
  font-size: .5rem;

  a {
    font-weight: bold;
    font-size: .8rem;
    cursor: pointer;
  }
}