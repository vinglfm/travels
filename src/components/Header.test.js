import enzyme, {shallow} from 'enzyme';
import ReactSixteenAdapter from 'enzyme-adapter-react-16';
enzyme.configure({ adapter: new ReactSixteenAdapter() });

import React from 'react';
import {Header} from './Header';

describe('<Header />', () => {
  let wrapper;
  
  beforeEach(() => {
    wrapper = shallow(<Header />);
  });

  it('renders SignIn', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
