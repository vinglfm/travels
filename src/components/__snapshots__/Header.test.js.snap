// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> renders SignIn 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <WithStyles(AppBar)
          title="Travels"
>
          <WithStyles(Toolbar)>
                    <WithStyles(Typography)
                              className={undefined}
                              color="inherit"
                              variant="title"
                    >
                              Travels
                    </WithStyles(Typography)>
                    <WithStyles(Button)
                              color="inherit"
                              onClick={undefined}
                    >
                              Sign Out
                    </WithStyles(Button)>
          </WithStyles(Toolbar)>
</WithStyles(AppBar)>,
        <AuthModal
          handleClose={[Function]}
          open={false}
/>,
      ],
      "className": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <WithStyles(Toolbar)>
            <WithStyles(Typography)
                        className={undefined}
                        color="inherit"
                        variant="title"
            >
                        Travels
            </WithStyles(Typography)>
            <WithStyles(Button)
                        color="inherit"
                        onClick={undefined}
            >
                        Sign Out
            </WithStyles(Button)>
</WithStyles(Toolbar)>,
          "title": "Travels",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <WithStyles(Typography)
                className={undefined}
                color="inherit"
                variant="title"
>
                Travels
</WithStyles(Typography)>,
              <WithStyles(Button)
                color="inherit"
                onClick={undefined}
>
                Sign Out
</WithStyles(Button)>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Travels",
                "className": undefined,
                "color": "inherit",
                "variant": "title",
              },
              "ref": null,
              "rendered": "Travels",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Sign Out",
                "color": "inherit",
                "onClick": undefined,
              },
              "ref": null,
              "rendered": "Sign Out",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "handleClose": [Function],
          "open": false,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <WithStyles(AppBar)
            title="Travels"
>
            <WithStyles(Toolbar)>
                        <WithStyles(Typography)
                                    className={undefined}
                                    color="inherit"
                                    variant="title"
                        >
                                    Travels
                        </WithStyles(Typography)>
                        <WithStyles(Button)
                                    color="inherit"
                                    onClick={undefined}
                        >
                                    Sign Out
                        </WithStyles(Button)>
            </WithStyles(Toolbar)>
</WithStyles(AppBar)>,
          <AuthModal
            handleClose={[Function]}
            open={false}
/>,
        ],
        "className": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <WithStyles(Toolbar)>
              <WithStyles(Typography)
                            className={undefined}
                            color="inherit"
                            variant="title"
              >
                            Travels
              </WithStyles(Typography)>
              <WithStyles(Button)
                            color="inherit"
                            onClick={undefined}
              >
                            Sign Out
              </WithStyles(Button)>
</WithStyles(Toolbar)>,
            "title": "Travels",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <WithStyles(Typography)
                  className={undefined}
                  color="inherit"
                  variant="title"
>
                  Travels
</WithStyles(Typography)>,
                <WithStyles(Button)
                  color="inherit"
                  onClick={undefined}
>
                  Sign Out
</WithStyles(Button)>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Travels",
                  "className": undefined,
                  "color": "inherit",
                  "variant": "title",
                },
                "ref": null,
                "rendered": "Travels",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Sign Out",
                  "color": "inherit",
                  "onClick": undefined,
                },
                "ref": null,
                "rendered": "Sign Out",
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "handleClose": [Function],
            "open": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
